protocol/messenger_backup_test.go

```diff
@@ -634,3 +634,48 @@ func (s *MessengerBackupSuite) TestBackupCommunities() {
 	s.Require().NotEmpty(lastBackup)
 	s.Require().Equal(clock, lastBackup)
 }
+
+func (s *MessengerBackupSuite) TestBackupKeycards() {
+	// Create bob1
+	bob1 := s.m
+	allKeycardsToSync := getKeycardsForTest()
+	for _, kp := range allKeycardsToSync {
+		addedKc, addedAccs, err := bob1.settings.AddMigratedKeyPairOrAddAccountsIfKeyPairIsAdded(*kp)
+		s.Require().NoError(err)
+		s.Require().Equal(true, addedKc)
+		s.Require().Equal(false, addedAccs)
+	}
+
+	// Create bob2
+	bob2, err := newMessengerWithKey(s.shh, bob1.identity, s.logger, nil)
+	s.Require().NoError(err)
+	_, err = bob2.Start()
+	s.Require().NoError(err)
+
+	// Backup
+	_, err = bob1.BackupData(context.Background())
+	s.Require().NoError(err)
+
+	// Wait for the message to reach its destination
+	_, err = WaitOnMessengerResponse(
+		bob2,
+		func(r *MessengerResponse) bool {
+			_, err := bob2.RetrieveAll()
```

Comment:
feel strange here since it already called `RetrieveAll` within `WaitOnMessengerResponse` 🤔 , in general, we return the condition that indicate we received the relate data we want

Reply:
I agree, but this is very steril test environment, don't expect issues, also this is how we're doing the same in other tests here.

Reply:
if you need "Wait for the message to reach its destination", some code like following should be better or you can check other things relate with variable r
```
WaitOnMessengerResponse(
		bob2,
		func(r *MessengerResponse) bool { return len(r.Messages()) > 0 },
		"no messages",
	)
```
@saledjenic

Reply:
@qfrank it doesn't work in that case, test fails.

Reply:
But it works without this line `_, err := bob2.RetrieveAll()`, this works:
```golang
_, err = WaitOnMessengerResponse(
		bob2,
		func(r *MessengerResponse) bool { return true },
		"no messages",
	)
```

Reply:
@qfrank I found a way to make the tests more reliable, please check it now.